src/xunit.performance.api/Profilers/Etw/SimpleTraceEventParser.cs

```diff
@@ -0,0 +1,378 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Microsoft.Diagnostics.Tracing;
+using Microsoft.Diagnostics.Tracing.Parsers;
+using Microsoft.Diagnostics.Tracing.Parsers.Clr;
+using Microsoft.Diagnostics.Tracing.Parsers.Kernel;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Microsoft.Xunit.Performance.Api.Profilers.Etw
+{
+    /// <summary>
+    /// Provides a simple interface to extract ETW process/modules data from an *.etl file.
+    /// </summary>
+    public sealed class SimpleTraceEventParser
+    {
+        /// <summary>
+        /// Gets profile data from the provided ScenarioInfo object
+        /// </summary>
+        /// <param name="scenarioInfo"></param>
+        /// <returns>A collection of the profiled processes for the given scenario.</returns>
+        /// <remarks>
+        /// Some assumptions:
+        ///     1. The scenario launches a single process, but itself can launch child processes.
+        ///     2. Process started/stopped within the ETW session.
+        /// </remarks>
+        public IReadOnlyCollection<Process> GetProfileData(ScenarioExecutionResult scenarioInfo)
+        {
+            var processes = new List<Process>();
+            Module tmpNtoskrnlModule = null;
+            var pmcSamplingIntervals = new Dictionary<int, long>();
+
+            Func<int, bool> IsOurProcess = (processId) => {
+                return processId == scenarioInfo.ProcessExitInfo.ProcessId || processes.Any(process => process.Id == processId);
+            };
+
+            using (var source = new ETWTraceEventSource(scenarioInfo.EventLogFileName))
+            {
+                if (source.EventsLost > 0)
+                    throw new Exception($"Events lost in trace '{scenarioInfo.EventLogFileName}'");
+
+                ////////////////////////////////////////////////////////////////
+                // Process data
+                var parser = new KernelTraceEventParser(source);
```

Comment:
I don't understand the asymmetry between start (which looks for parent) and stop (which doesn't). Can you explain?

Reply:
No, I should move this into the `Func` object. This is a left over from cleanup.

---
In reply to: [145197586](https://github.com/Microsoft/xunit-performance/pull/242#discussion_r145197586) [](ancestors = 145197586)